phpstan.neon

```diff
@@ -1,5 +1,10 @@
 parameters:
     level: 10
+    phpVersion: 80402
```

Comment:
Without this setting, PHPStan would report `BcMath\Number` as an unknown class and we would lose type coverage for the new `NumberComparator` class.

Reply:
I just updated PHPStan to version 2.1.1 on `main`. Maybe that solves this?

Reply:
No. PHPStan 2.0.1 does already know about the `Number` class, so upgrading to 2.1.1 doesn't change anything for me.

The problem is that PHPStan needs a target PHP version for its analysis. If we don't set it here, it will guess it from the composer.json file and assume PHP 8.2.0 because of this setting:

https://github.com/sebastianbergmann/comparator/blob/07ba41ddc42632b0a78800bca6a6ce003bf60663/composer.json#L41-L43

The `Number` class simply does not exist on PHP 8.2 yet, so PHPStan technically correct when complaining about this.

Reply:
BC Number support is tracked here: https://github.com/phpstan/phpstan/issues/12099

Reply:
@staabm That's for the overloaded operators only which is a different issue. Once the issue you've linked is resolved, I can remove the one ignored error from the PHPStan configuration file.